rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -207,15 +207,15 @@ namespace rsx
 		{
 			if (rsx->in_begin_end)
 			{
-				rsx->append_array_element(arg & 0xFFFF);
-				rsx->append_array_element(arg >> 16);
+				rsx->append_array_element(se_storage<u16>::swap(arg & 0xFFFF));
+				rsx->append_array_element(se_storage<u16>::swap(arg >> 16));
```

Comment:
The indices are like this in the argument: 0x00010002.
Then they get unpacked to u16: 0x0001 and 0x0002.
Then they are endian swapped as u16 by this code: 0x0100 and 0x0200.
Then they get pushed to an u32 vector: 0x00000100 and 0x00000200.
Then they get interpreted by the drawing code, which swaps the endianess again: 0x00010000 and 0x00020000.

This is how I see it and why the value needs to be initially swapped as a 32 bit value.